### PR TITLE
add openwebtext2

### PR DESCRIPTION
**Pull request by richarddwang**
openwebtext2 is part of EleutherAI/The Pile, but AFAIK, The Pile dataset blend all sub datasets together thus we are not able to use just one of its sub dataset from The Pile data. So I create an independent dataset using The Pile preliminary components.

When I was creating dataset card. I found there is room for creating / editing dataset card. I've made it an issue. #2797

Also I am wondering whether the import of The Pile dataset is actively undertaken (because I may need it recently)? #1675

### CODE REVIEW

**Comment by richarddwang** [comment]
It seems we need to `pip install jsonlines` to pass the checks ?

**Comment by lhoestq** [comment]
Hi ! Do you really need `jsonlines` ? I think it simply uses `json.loads` under the hood.

Currently the test are failing because `jsonlines` is not part of the extra requirements `TESTS_REQUIRE` in setup.py

So either you can replace `jsonlines` with a simple for loop on the lines of the files and use `json.loads`, or you can add `TESTS_REQUIRE` to the test requirements (but in this case users will have to install it as well).

**Comment by richarddwang** [comment]
Thanks for your suggestion. I've knew `io` and json lines format better and changed `jsonlines` to just `readlines`.

File: datasets/openwebtext2/README.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+- `en`
+
+## Dataset Structure
+
+### Data Instances
+
+```
+{'title': 'title',
+'text': 'some text',
+'reddit_scores': [6],}
```
Comment (lhoestq) [inline datasets/openwebtext2/README.md:88]: This example is too long and was cropped

```suggestion
This example was too long and was cropped:

{'title': Xiaomi Mi Note 10 Gearbest Coupon Promo Code [6+128GB] [France Warehouse],
'text': '27% off Xiaomi Mi Note 10 (CC9 Pro) 108MP Penta Camera Mobile Phone Global Version Online Smartphone – Black Gearbest Coupon Promo Code\n\nGearbest Coupon Price :$439.99\n\nRegular Price : $603.19 Your Save : $163.20 Coupon Limit: 100 times Warehouse: France Expires : September 30, 2020 Coupon Valid for...',
'reddit_scores': [6],}
```

File: datasets/openwebtext2/README.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+```
+
+### Data Fields
+
+- `title`
+- `text`
+- `reddit_scores`
```
Comment (lhoestq) [inline datasets/openwebtext2/README.md:95]:
```suggestion
- `title`: title of the web page
- `text`: text content of the web page
- `reddit_scores`: scores of the reddit submissions that mention this web page, as a list of integers
```

File: datasets/openwebtext2/README.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  - [Additional Information](#additional-information)
+    - [Dataset Curators](#dataset-curators)
+    - [Licensing Information](#licensing-information)
+    - [Citation Information](#citation-information)
+    - [Contributions](#contributions)
+
+## Dataset Description
+
+- **Homepage:** https://openwebtext2.readthedocs.io/en/latest/
+- **Repository:** [Needs More Information]
```
Comment (lhoestq) [inline datasets/openwebtext2/README.md:61]:
```suggestion
- **Repository:** [GitHub](https://github.com/EleutherAI/openwebtext2)
```

File: datasets/openwebtext2/README.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+}
+```
+
+### Contributions
+
+[researcher2](https://github.com/researcher2) Wrote much of this code, with inspiration and some straight copying of the scraping code found [here](https://github.com/yet-another-account/openwebtext/).<br/>
+[sdtblck](https://github.com/sdtblck/) kindly put together the Colab notebook, and performed a chunk of the scraping. <br/>
+[leogao2](https://github.com/leogao2/) provided overall design guidance, lm_dataformat, and performed another chunk of scraping. <br />
+[Colaboratory](https://colab.research.google.com/) VMs helped us with about 10% of our overall scraping. <br />
+[The Eye](http://the-eye.eu/) host our processed datasets.<br />
```
Comment (lhoestq) [inline datasets/openwebtext2/README.md:174]:
```suggestion
[Colaboratory](https://colab.research.google.com/) VMs helped with about 10% of our overall scraping. <br />
[The Eye](http://the-eye.eu/) host the processed datasets.<br />
```

File: datasets/openwebtext2/README.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+[Read The Docs](https://readthedocs.org/) host our documentation.<br />
+
+[@richarddwang](https://github.com/richarddwang) adds this dataset to HF/datasets.
```
Comment (lhoestq) [inline datasets/openwebtext2/README.md:177]:
```suggestion
[@richarddwang](https://github.com/richarddwang) added this dataset to HF/datasets.
```

**Review by lhoestq — APPROVED** [review]
Awesome thanks !

I'm also doing some minor modifications and we can merge :)

**Pull request merged**